src/components/TransactionBlock/WalletTransactionBlock.tsx

```diff
@@ -0,0 +1,737 @@
+                            <Text size={12} color="#191726" regular>{`on ${chainAsset.title}`}</Text>
+                            <Text size={12} color="#191726" regular>{`${
+                              formatAmountDisplay(ethers.utils.formatUnits(asset.balance, asset.decimals)) || 0
+                            } ${asset.symbol}`}</Text>
+                          </ListItemLine>
+                        </ListItemDetails>
+                      </ListItem>
+                    );
+                  })}
+                </ChainBlockList>
+              )}
+            </ChainBlock>
+          );
+        })}
+
+      {!showAllChains && displayAssets?.length > 0 && (
+        <ChainBlock>
+          <ChainBlockList>
+            {displayAssets?.map((asset) => {
+              const chain = supportedChains.find((item) => item.chainId === asset.chainId);
+              if (!chain) return;
+
+              return (
+                <ListItem>
+                  <ListItemIconWrapper>
+                    <RoundedImage url={asset.logoURI} size={32} title={asset.name} noMarginRight />
+                    <AssetChainIcon src={chain.iconUrl} title={chain.title} />
+                  </ListItemIconWrapper>
+                  <ListItemDetails>
+                    <ListItemLine>
+                      <Text size={14} color="#191726" medium>{`${asset.symbol}・$${
+                        asset.assetPriceUsd?.toFixed(2) || 0
+                      }`}</Text>
+                      <Text size={14} color="#191726" medium>{`$${asset.balanceWorthUsd?.toFixed(2) || 0}`}</Text>
+                    </ListItemLine>
+
+                    <ListItemLine>
+                      <Text size={12} color="#191726" regular>{`on ${chain.title}`}</Text>
+                      <Text size={12} color="#191726" regular>{`${
+                        formatAmountDisplay(ethers.utils.formatUnits(asset.balance, asset.decimals)) || 0
+                      } ${asset.symbol}`}</Text>
+                    </ListItemLine>
+                  </ListItemDetails>
+                </ListItem>
+              );
+            })}
+          </ChainBlockList>
+        </ChainBlock>
+      )}
+    </>
+  );
+};
+
+export default WalletTransactionBlock;
+
+const Title = styled.h3`
+  margin: 0 0 18px;
+  padding: 0;
+  font-size: 20px;
+  color: ${({ theme }) => theme.color.text.cardTitle};
+  font-family: 'PTRootUIWebMedium', sans-serif;
+`;
+
+// Buttons
+const ButtonRow = styled.div`
+  display: flex;
+  flex: 1;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  margin: 0 0 18px;
+`;
+
+const ActionButtonWrapper = styled.div<{ disabled?: boolean }>`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  flex-direction: column;
+  cursor: pointer;
+
+  ${({ disabled }) =>
+    !disabled
+      ? `&:hover {
+    span {
+      color: #ff7733;
+    }
+    div {
+      opacity: 0.5;
+    }
+  }`
+      : 'opacity: 0.5'};
```

Comment:
this reads hard, maybe add {disabled && ``} {!disabled && ``}

```suggestion
  ${({ disabled }) =>
    !disabled
      ? `&:hover {
    span {
      color: #ff7733;
    }
    div {
      opacity: 0.5;
    }
  }`
      : 'opacity: 0.5'};
```

Reply:
yeah sure can do